apps/passport-client/components/screens/ProveScreen/ProveScreen.tsx

```diff
@@ -29,6 +29,19 @@ export function ProveScreen() {
   const request = JSON.parse(params.get("request")) as PCDGetRequest;
   const screen = getScreen(request);
 
+  useEffect(() => {
+    if (process.env.NODE_ENV !== "production") {
+      // NB: Telegram bot doesn't support localhost url and we had to use 127.0.0.1 instead
+      // We redirect back to localhost in development mode
+      if (window.location.hostname === "127.0.0.1") {
+        window.location.href = window.location.href.replace(
+          "://127.0.0.1",
+          "://localhost"
+        );
```

Comment:
nit: can we use [window.location.replace](https://developer.mozilla.org/en-US/docs/Web/API/Location/replace) instead? So that it doesn't leave the 127.0.0.1 url in the browser history.

cc @ichub if you have strong opinions

```suggestion
        window.location.replace(
          window.location.href.replace("://127.0.0.1", "://localhost")
        );
```